### ISSUE TITLE
cppcheck errors

### ISSUE THREAD

**Issue by onlyjob** [moosefs/moosefs#2]
`cppcheck` found multiple errors, notably

~~~~
[mfschunkserver/masterconn.c:1295]: (error) Common realloc mistake: 'readbuff' nulled but not freed upon failure
[mfsmaster/matoclserv.c:5678]: (error) Common realloc mistake: 'readbuff' nulled but not freed upon failure
[mfsmaster/matoclserv.c:3855]: (error) Uninitialized variable: cs_data
[mfsmaster/matocsserv.c:2137]: (error) Common realloc mistake: 'readbuff' nulled but not freed upon failure
[mfsmaster/matomlserv.c:663]: (error) Common realloc mistake: 'readbuff' nulled but not freed upon failure
[mfsmetalogger/masterconn.c:754]: (error) Common realloc mistake: 'readbuff' nulled but not freed upon failure
[mfsmount/mfs_fuse.c:181]: (error) Common realloc mistake: 'sinfo_tab' nulled but not freed upon failure
[mfsmount/mfs_fuse.c:266]: (error) Common realloc mistake: 'dirbuf_tab' nulled but not freed upon failure
[mfsmount/mfs_fuse.c:351]: (error) Common realloc mistake: 'finfo_tab' nulled but not freed upon failure
[mfsmount/readdata.c:1653]: (error) Common realloc mistake: 'ranges' nulled but not freed upon failure
[mfsmount/readdata.c:1666]: (error) Common realloc mistake: 'ranges' nulled but not freed upon failure
~~~~

as well as the following warnings:

~~~~
[mfscommon/cfg.c:205]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                               
[mfscommon/charts.c:1316]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1319]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1322]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1325]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1334]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1336]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1356]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1362]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1373]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1380]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1397]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1399]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/charts.c:1404]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/conncache.c:82]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/conncache.c:101]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:111]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:118]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:131]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:163]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:171]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:173]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:175]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:183]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/conncache.c:184]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                         
[mfscommon/delayrun.c:78]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/delayrun.c:81]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/delayrun.c:106]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:108]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:112]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:116]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:121]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:125]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:133]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:135]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:139]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:142]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:144]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:145]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:146]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:147]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:164]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:167]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:168]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:170]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:171]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:178]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:180]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:182]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/delayrun.c:184]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/lwthread.c:89]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/lwthread.c:90]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/lwthread.c:99]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                           
[mfscommon/lwthread.c:104]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/lwthread.c:106]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                          
[mfscommon/main.c:185]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:193]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:201]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:209]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:217]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:225]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:233]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:242]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:250]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:265]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:719]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                              
[mfscommon/main.c:1063]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1065]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1067]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1075]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1140]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1143]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1178]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/main.c:1285]: (warning) %u in format string (no. 1) requires 'unsigned int' but the argument type is 'signed int'.                                             
[mfscommon/pcqueue.c:55]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.                                            
[mfscommon/pcqueue.c:65]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:67]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:68]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:75]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:76]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:77]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:83]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:84]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:85]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:87]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:94]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:97]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:101]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:104]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:110]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:112]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:119]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:121]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:128]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:130]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:137]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:143]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:151]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:157]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:160]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:167]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:170]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:181]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:184]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:191]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:194]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:199]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:205]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:216]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:219]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:226]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:229]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:232]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:256]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:259]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:279]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:281]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:305]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/pcqueue.c:308]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfscommon/strerr.c:538]: (warning) %u in format string (no. 2) requires 'unsigned int' but the argument type is 'signed int'.
[mfsmaster/filesystem.c:5251]: (warning) The array 'chunkcount' is too small, the function 'fsnodes_checkfile' expects a bigger one.
[mfsmaster/restore.c:266]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:334]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:415]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:424]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:454]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:602]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:720]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:730]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:748]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:806]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:920]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:955]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:999]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:1118]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:1139]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:1161]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmaster/restore.c:1187]: (warning) Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?
[mfsmount/main.c:728]: (warning) %d in format string (no. 1) requires 'int' but the argument type is 'unsigned int'.
~~~~

**Comment by acid-maker** [comment]
Which version of cppcheck did you use?

I've used 1.72 and it gave me totally different messages (only one was really important, and this one was not on your list - strange).

My full invocation:

    cppcheck --quiet --std=c99 --std=posix --platform=unix64 -Imfscommon --force .
    
My output:

[mfscommon/portable.h:186]: (error) Resource leak: s
[mfscommon/portable.h:212]: (error) Resource leak: tmp_sock
[mfscommon/main.c:1063]: (error) Return value of allocation function 'open' is not stored.
[mfscommon/main.c:1075]: (error) Return value of allocation function 'open' is not stored.
[mfsmaster/matoclserv.c:3855]: (error) Uninitialized variable: cs_data
[mfsmaster/metadata.c:1240]: (error) Memory leak: name
[mfsmaster/metadata.c:1227]: (error) Memory leak: name
[mfsmaster/metadata.c:1232]: (error) Memory leak: name

portable.h:

This is such code:

        if ((s = socket(AF_INET, SOCK_STREAM, 0)) == INVALID_SOCKET) {
                printf("socket error: %u\n",GetLastError());
                return -1;
        }

I don't know why cppcheck thinks that there is resource leak. "return -1" is invoked without closing socket 's' but only when there is no socket 's' (s == INVALID_SOCKET) !!! Cppcheck is a great tool but sometimes can produce false negatives. This looks like one.


main.c:

This is open of "/dev/null" after closing descriptors 0,1 and 2 (classic daemonizing procedure).

The line looks like this:

       sassert(open("/dev/null", O_RDWR, 0)==STDIN_FILENO);

Value is not stored because it always should be zero and returned value is properly checked. Looks like another false negative.

matoclserv.c:

The line looks like this:

      memcpy(ptr,cs_data,count*7);

and cs_data is not initialized (but points to valid memory block) only when count is zero, so this is not mistake, but I've fixed that to silent this warning by adding "if (count>0) { ... }" - now the code looks better.

metadata.c:

This is true leak. I've fixed that.

All other "errors" present in your output are produced by code which looks like this (one example):

  sinfo_tab = realloc(sinfo_tab,sizeof(sinfo*)*sinfo_size);
  passert(sinfo_tab);

Formally construction `a = realloc(a,...)` is wrong because when realloc returns NULL you should free previous pointer, so this code should look more like this:

```
  na = realloc(a,...);
  if (na==NULL) {
    free(a);
  }
  a = na;

```
But since in my code I always do `abort()` when returned pointer is NULL then I do not need to free previous pointer. Likely my version of cppcheck is able to "see" this "abort" and do not produce this false negatives.

Nevertheless during checking your post I've found other error (in metadata.c) reported by my cppcheck, so thanks a lot for reporting this issue :)

I've also corrected all warnings.

**Comment by onlyjob** [comment]
> Which version of cppcheck did you use?

1.72 from Debian "testing".

> I've used 1.72 and it gave me totally different messages

Try something like `cppcheck . --force --enable=warning --quiet` from source tree root.

> (only one was really important, and this one was not on your list - strange).

That could be explained by my poor copy-pasting skills. ;)

> This is true leak. I've fixed that.

Where?? I don't see relevant commit. When you fix it please mention commit ID.

> so thanks a lot for reporting this issue :)

The pleasure is mine. :) Thanks for using `cppcheck`. Routine cpp-checks  is a good practice.

> I've also corrected all warnings.

Fantastic. When can we expect corresponding commits to repository?

**Comment by acid-maker** [comment]
Now I see those "realloc" errors. In my checking script I've some extra options "--std=c99 --std=posix --platform=unix64". One of them probably silenced realloc (and some other) errors.

Now I see some other warnigs/errors to check. Thanks again. I'll add these frees after reallocs - just to silence cppcheck. Even if this is a false negative still it is good to silence that because it will prevent future checks.

You will see commit soon. I've changed that only in my private code repository. Today we will commit 3.0.74 with mentioned changes. I now that you don't like my "bulk" commits, but as for now it is much easier for me to work in that way. Maybe I'll change that in the future. I still need more practice with git to do that.

**Comment by onlyjob** [comment]
> ... bulk" commits, but as for now it is much easier for me to work in that way.

Please do not be shy and just stop squashing individual commits. By doing so you destroy important advantages of public repositories. Individual commits are useful for backporting, bug squashing, bisecting etc.

> Maybe I'll change that in the future. I still need more practice with git to do that.

I strongly encourage you to practice that ASAP. It is easier than you think. ;)

**Comment by onlyjob** [comment]
Vcs with only bulk commits is no more useful than collection of tarball releases.
What is the purpose of Vcs then?

**Comment by acid-maker** [comment]
Yes. I agree with you. I'll try to change my habits :)

**Comment by onlyjob** [comment]
Thanks. :)

**Comment by acid-maker** [comment]
I've pushed my first partial commit - which fixes this cppcheck warnings/errors :)

**Comment by onlyjob** [comment]
Nice. :) If you include "#2" ("#" followed by the bug number) to commit message there will be automatic reference in corresponding bug report. I had a look at 7d48b022 -- ideally it would address only warnings/errors (as its description says) without touching NEWS, Makefiles, changing version etc. Version bump and NEWS summary normally do not belong to the same commit that addresses compilation warnings.

**Comment by oxide94** [comment]
Fixed in 7d48b02.
